components/util/__test__/Anchors.test.tsx

```diff
@@ -0,0 +1,50 @@
+import { create as render, act } from "react-test-renderer";
+import { createElement } from "react";
+import Anchors, { useAnchors } from "../Anchors";
+
+const mockPath = "/";
+jest.mock("next/router", () => ({
+  useRouter() {
+    return {
+      route: mockPath,
+      pathname: mockPath,
+      query: "",
+      asPath: mockPath,
+    };
+  },
+}));
+
+function DebugAnchors() {
+  return <pre>{JSON.stringify(useAnchors().anchors, null, 2)}</pre>;
+}
+
+function MockMDXCreateElement(props: {
+  children: string;
+  mdxType: string;
+  id?: string;
+}) {
+  return createElement(props.mdxType, {
+    children: props.children,
+    id: props.id,
+  });
+}
+
+test("Anchors", async () => {
+  let tree;
+  act(() => {
```

Comment:
Since `Anchors.Set` performs an update on the Anchors context, we need to wrap this in `act(() => ...)`. Else `DebugAnchors` would show an empty array for the anchors.

https://reactjs.org/docs/test-renderer.html#testrendereract